### PR TITLE
RPC errors

### PR DESCRIPTION
**Pull request by johnstonematt**

### CODE REVIEW

File: cmd/solana_exporter/slots.go [hunk 1]
```diff
@@ -331,6 +334,14 @@ func (c *SlotWatcher) fetchAndEmitSingleFeeReward(
 ) error {
 	block, err := c.client.GetBlock(ctx, rpc.CommitmentConfirmed, slot)
 	if err != nil {
+		var rpcError *rpc.RPCError
+		if errors.As(err, &rpcError) {
+			// this is the error code for slot was skipped:
+			if rpcError.Code == -32007 && strings.Contains(rpcError.Message, "skipped") {
+				klog.Infof("slot %v was skipped, no fee rewards.", slot)
```
Comment (SEJeff) [inline cmd/solana_exporter/slots.go:341]: Warnf maybe?

File: cmd/solana_exporter/slots.go [hunk 2]
```diff
@@ -331,6 +334,14 @@ func (c *SlotWatcher) fetchAndEmitSingleFeeReward(
 ) error {
 	block, err := c.client.GetBlock(ctx, rpc.CommitmentConfirmed, slot)
 	if err != nil {
+		var rpcError *rpc.RPCError
+		if errors.As(err, &rpcError) {
+			// this is the error code for slot was skipped:
+			if rpcError.Code == -32007 && strings.Contains(rpcError.Message, "skipped") {
```
Comment (SEJeff) [inline cmd/solana_exporter/slots.go:340]: Can you make that a constant please? Also, where did you get that rpc error code from? The solana source or some docs? If from some docs, can you link them in a comment 🙏

**Review by SEJeff — APPROVED** [review]

File: cmd/solana_exporter/slots.go (hunk 2 above)
Comment (johnstonematt) [reply cmd/solana_exporter/slots.go:340]: No docs, just ran the call on a block that was skipped and looked at the code. Can make it a constant tho

File: cmd/solana_exporter/slots.go (hunk 1 above)
Comment (johnstonematt) [reply cmd/solana_exporter/slots.go:341]: I mean, this part of the exporter is not designed for tracking skipped slots so in theory this is not necessarily a bad thing. For instance, we might want to track the fee rewards of Triton 1 to see if they are doing something we aren't (or just a handful of validator benchmarks), so does it make sense for us to get a warning when they have a skipped slot?

File: cmd/solana_exporter/slots.go (hunk 2 above)
Comment (SEJeff) [reply cmd/solana_exporter/slots.go:340]: [Here you go](https://github.com/anza-xyz/agave/blob/489f483e1d7b30ef114e0123994818b2accfa389/rpc-client-api/src/custom_error.rs#L17) sir!

File: pkg/rpc/responses.go [hunk 3]
```diff
@@ -6,10 +6,15 @@ import (
...
+	RPCError struct {
+		Message string `json:"message"`
+		Code    int64  `json:"code"`
+	}
+
 	response[T any] struct {
 		jsonrpc string
 		Result  T        `json:"result"`
-		Error   rpcError `json:"error"`
+		Error   RPCError `json:"error"`
```
[the remark this answers is not in the document]
Comment (SEJeff) [reply pkg/rpc/responses.go:17]: @GranderStark feel free to propose this in #33 if youd like.

**Pull request merged**